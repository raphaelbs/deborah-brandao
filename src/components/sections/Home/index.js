import React from 'react';
import { Grid, Typography, useTheme, useMediaQuery } from '@material-ui/core';

import Image from './Image';

function Home() {
  const theme = useTheme();
  const isDesktop = useMediaQuery(theme.breakpoints.up('md'));

  const nameHeight = '140px';
  const imageHeight = `calc(100% - ${nameHeight})`;

  const nameStyles = {
    maxHeight: isDesktop ? 'inherit' : nameHeight,
    height: isDesktop ? '100%' : 'auto',
  };
  const nameAlign = isDesktop ? 'center' : 'left';

  const imageStyles = {
    height: isDesktop ? '100%' : imageHeight,
    border: '1px solid grey',
    padding: 0,
  };

  return (
    <Grid container spacing={0} className="viewport">
      <Grid
        item
        container
        xs={12}
        md={4}
        style={nameStyles}
        alignContent="center"
      >
        <Typography variant="h1" align={nameAlign}>
          Deborah Brand√£o
        </Typography>
      </Grid>
      <Grid item xs={12} md={8} style={imageStyles}>
        <Image />
      </Grid>
    </Grid>
  );
}

export default Home;
