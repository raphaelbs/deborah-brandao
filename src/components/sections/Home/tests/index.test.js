import React from 'react';
import renderer from 'react-test-renderer';

import useMediaQuery from '@material-ui/core/useMediaQuery';

jest.mock('@material-ui/core/useMediaQuery');

jest.mock('gatsby', () => ({
  graphql: jest.fn(() => ({
    placeholderImage: {
      childImageSharp: {
        fluid: 'image-source',
      },
    },
  })),
  useStaticQuery: jest.fn(param => param),
}));

import Home from '../index';

describe('Test home section', () => {
  it('should return the mobile html', () => {
    useMediaQuery.mockReturnValue(false);

    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should return the desktop html', () => {
    useMediaQuery.mockReturnValue(true);

    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
