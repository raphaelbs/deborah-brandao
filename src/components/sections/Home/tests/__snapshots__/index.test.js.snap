// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test home section should return the desktop html 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
  style={
    Object {
      "height": "100%",
      "margin": 0,
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-content-xs-center MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
    style={
      Object {
        "height": "100%",
        "maxHeight": "inherit",
      }
    }
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter"
    >
      Deborah Brandão
    </h1>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
    style={
      Object {
        "border": "1px solid grey",
        "height": "100%",
        "padding": 0,
      }
    }
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "height": "100%",
          "maxHeight": null,
          "maxWidth": null,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        aria-hidden={true}
        style={
          Object {
            "paddingBottom": "NaN%",
            "width": "100%",
          }
        }
      />
      <picture>
        <source />
        <img
          alt=""
          loading="lazy"
          onLoad={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Test home section should return the mobile html 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
  style={
    Object {
      "height": "100%",
      "margin": 0,
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-content-xs-center MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
    style={
      Object {
        "height": "auto",
        "maxHeight": "140px",
      }
    }
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 MuiTypography-alignLeft"
    >
      Deborah Brandão
    </h1>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
    style={
      Object {
        "border": "1px solid grey",
        "height": "calc(100% - 140px)",
        "padding": 0,
      }
    }
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "height": "100%",
          "maxHeight": null,
          "maxWidth": null,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        aria-hidden={true}
        style={
          Object {
            "paddingBottom": "NaN%",
            "width": "100%",
          }
        }
      />
      <picture>
        <source />
        <img
          alt=""
          loading="lazy"
          onLoad={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </div>
</div>
`;
